ruby-challenges.rb

```diff
@@ -6,27 +6,73 @@
 beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
 
 letter_o = 'o'
-# Expected output: ['coffee', 'soda water']
+
 letter_t = 't'
-# Expected output: ['tea', 'water', 'soda water']
 
+# gonna def my function as liquid 
+# set two parameter for the letters and beverages_array
+# use .select to value then .include to see if it contains our specific string 
+def liquid (array, letter) 
+    array.select do |value| (value).include?(letter)
+    end
+end
+p liquid(beverages_array, letter_o )
+# Expected output: ['coffee', 'soda water']
+p liquid(beverages_array, letter_t)
+# Expected output: ['tea', 'water', 'soda water']
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
+# def worst_states as my function
+# one parameter for the array of us_states
+# .values used for getting the array values in alphabetical order
+# flatten to get rid of nested arrays 
+# sort will be the method to take the values and palce them to recive the array in alphabetical order
+def worst_states (array)
```

Comment:
I will allow this only because, the greatest State ever founded, New York is not listed but tread carefully Christian 😂